test/setup/makeTestSettings.ts

```diff
@@ -0,0 +1,124 @@
+import { SevereServiceError } from 'webdriverio';
+import { Frameworks } from '@wdio/types';
+import WikibaseApi from 'wdio-wikibase/wikibase.api.js';
+import { defaultFunctions as defaultFunctionsInit } from '../helpers/default-functions.js';
+import loadEnvFiles from './loadEnvFiles.js';
+import loadLocalDockerImage from './loadLocalDockerImage.js';
+import { saveScreenshot } from 'wdio-mediawiki';
+import TestSettings, {
+	TestEnvSettings,
+	TestHooks,
+	TestRunnerSettings,
+	TestSuiteSettings
+} from '../types/TestSettings.js';
+
+export const defaultTestSettings = {
+	envFiles: [
+		'../variables.env',
+		'./test-services.env',
+		'../local.env'
+	],
+	composeFiles: [
+		'suites/docker-compose.yml'
+	],
+	waitForUrls: (): string[] => {
+		return ( [
+			`${testEnv.vars.WIKIBASE_URL}/wiki/Main_Page`,
+			`${testEnv.vars.WDQS_URL}/bigdata/namespace/wdq/sparql`,
+			testEnv.vars.WDQS_FRONTEND_URL
```

Comment:
Isn't using `testEnv` in `makeTestSettings` circular?

Reply:
They are within a function definition and not evaluated until the 'testEnv` is created and running and we check if the services are up in `TestEnv#up` (which is ran by WDIO `onPrepare` hook). The functions are just being defined, not evaluated. It's the same as using `browser.addCommand` in `default-functions.ts`, as `browser` doesn't exist either until the `before` hook is ran.